app/scripts/lib/validate.js

```diff
@@ -103,6 +103,47 @@ define(function (require, exports, module) {
       }
 
       return password.length >= Constants.PASSWORD_MIN_LENGTH;
+    },
+
+    /**
+     * Check whether data matches schema, depth-first.
+     * Compares using Object.prototype.toString.
+     */
+    isDataValid: function isDataValid (data, schema) {
+      if (! schema) {
+        return false;
+      }
+
+      if (typeof schema === 'string') {
+        if (schema[0] === '?') {
+          if (data === null || data === undefined) {
+            return true;
+          }
+
+          schema = schema.substr(1);
+        }
+
+        return Object.prototype.toString.call(data) === '[object ' + schema + ']';
+      }
+
+      if (! data) {
+        return false;
+      }
+
+      return Object.keys(data).reduce(function (isValid, key) {
```

Comment:
You might be able to use underscore's [some](http://underscorejs.org/#some) function here, it short circuits as soon as a match is found, and would eliminate the next few lines.